seqr/views/apis/project_api_tests.py

```diff
@@ -93,6 +80,44 @@ def test_create_update_and_delete_project(self):
         new_project = Project.objects.filter(name='new_project')
         self.assertEqual(len(new_project), 0)
 
+    @mock.patch('seqr.views.utils.permissions_utils.ANALYST_PROJECT_CATEGORY', 'analyst-projects')
+    @mock.patch('seqr.views.utils.permissions_utils.ANALYST_USER_GROUP', 'analysts')
+    @mock.patch('seqr.views.utils.permissions_utils.PM_USER_GROUP', 'project-managers')
+    def test_update_project(self):
+        update_project_url = reverse(update_project_handler, args=[PROJECT_GUID])
+        self.check_manager_login(update_project_url)
+
+        project = Project.objects.get(guid=PROJECT_GUID)
+        expected_workspace_name = project.workspace_name
+        self.assertEqual(project.genome_version, '37')
+
+        response = self.client.post(update_project_url, content_type='application/json', data=json.dumps(
+            {'description': 'updated project description', 'genomeVersion': '38', 'workspaceName': 'test update name'}
+        ))
+        self.assertEqual(response.status_code, 200)
+        updated_json = response.json()['projectsByGuid'][PROJECT_GUID]
+        self.assertEqual(updated_json['description'], 'updated project description')
+        # genome version and workspace should not update
+        self.assertEqual(updated_json['genomeVersion'], '37')
+        self.assertEqual(updated_json['workspaceName'], expected_workspace_name)
+        updated_project = Project.objects.get(guid=PROJECT_GUID)
+        self.assertEqual(updated_project.description, 'updated project description')
+        self.assertEqual(updated_project.genome_version, '37')
+        self.assertEqual(updated_project.workspace_name, expected_workspace_name)
+
+        # test consent code
+        response = self.client.post(update_project_url, content_type='application/json', data=json.dumps(
+            {'consentCode': 'G'}
+        ))
+        self.assertEqual(response.status_code, 403)
```

Comment:
You may want to test the error message as well.

Reply:
So the way our middleware works for security reason is any 403 permission denied errors we log the message but only return the error "Permission Denied" to the user, we don't return the message. That code is tested pretty extensively already so I don't think adding another test for it here is valuable
https://github.com/broadinstitute/seqr/blob/master/seqr/utils/middleware.py#L109